[Tan Jovan] iP

jovantanyk opened the pull request:
# Paimon: Your Streamlined Task Management Chatbot

**Paimon** is an innovative chatbot designed to **streamline all your task-related needs**. With Paimon, managing your tasks becomes effortless and efficient. Our goal is to provide a user-friendly interface that helps you keep track of everything from simple to-dos to complex event planning.

## Current Support

Paimon currently supports a variety of tasks to help organize your day-to-day activities:

- **Deadline Tasks**: Keep track of your important deadlines with ease.
- **Event Tasks**: Plan your events without missing a beat.
- **Todo Tasks**: Manage your daily tasks to stay on top of your game.

## Cool Commands to Try Out

We've developed a set of commands that make task management a breeze:

1. **Delete task**: Easily remove tasks that are no longer needed.
2. **Find task**: Quickly locate any task with a simple command.
3. **Mark/Unmark task**: Keep track of your task completion status.
4. **List all tasks**: Get an overview of all your tasks at a glance.

## Upcoming Features

Our team is constantly working to improve Paimon with new features. Here's what we have planned:

- [x] **Basic Commands**: Core functionality for task management.
- [x] **Delete Command**: Enhanced ability to remove tasks.
- [x] **Find Command**: Improved search capabilities.
- [x] **JUnit Testing**: Ensuring reliability and performance.
- [ ] **GUI Interface**: A graphical user interface for an intuitive experience.
- [ ] **Checkstyle**: Code quality and style checks for maintainability.

Stay tuned for these exciting updates as we continue to enhance Paimon's capabilities!

```
    public static void main(String[] args) {
    }
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

jovantanyk committed:
Level-0

jovantanyk committed:
Level-1: Echo

jovantanyk committed:
Level-2: Add, List

jovantanyk committed:
Renamed Main Class

jovantanyk committed:
Level 3: Mark as Done

jovantanyk committed:
Level 4: ToDos, Events, Deadlines

jovantanyk committed:
A-TextUiTesting

jovantanyk committed:
Level-5: Handle Errors

jovantanyk committed:
Adjusted Chat Formatting

jovantanyk committed:
Level-6: Delete

jovantanyk committed:
Level-7: Implemented Save/Load feature

Included example paimon.txt file for reference

jovantanyk committed:
Merge Level-7 into master branch

jovantanyk committed:
A-Packages: Shifted tasks to its own package

Also implemented more OOP for TaskLists in the form of encapsulation and abstraction

jovantanyk committed:
Merged A-Packages into main

jovantanyk committed:
Added basic date checking

Alteqa commented on src/main/java/Paimon.java:
```diff
@@ -0,0 +1,180 @@
+            CommandParser parser = new CommandParser(userInput);
+
+            switch (parser.getType()) {
+                case "bye":
```
case should be at the same indentation as switch

zhekaiii replied:
You can change styling in your IntelliJ settings so your autoformatter will let your case have the same indentation as switch.

Alteqa commented on src/main/java/FileHandler.java:
```diff
@@ -0,0 +1,79 @@
+
+    public static void saveTaskList(TaskList list) {
+        try {
+            FileWriter w = new FileWriter(FILE_PATH);
```
variable name should be descriptive.

Alteqa commented on src/main/java/task/TodoTask.java:
```diff
@@ -0,0 +1,20 @@
+    }
+
+    @Override
+    public String toFileString() {
```
naming should be self-explanatory

Alteqa commented on src/main/java/task/EventTask.java:
```diff
@@ -0,0 +1,29 @@
+
+    @Override
+    public String getTask() {
+        if (isDone) {
```
be consistent with using this.isDone()

Alteqa reviewed (commented):
Overall code looks good with a few very minor errors.

willowisp01 commented on src/main/java/CommandParser.java:
```diff
@@ -0,0 +1,84 @@
+    public String[] parseInput() throws ChatException {
+        switch (this.type) {
+            case "todo":
+                String regex1 = "^(\\w+)(\\s)(.*)";
```
As mentioned [here](https://nus-cs2103-ay2324s2.github.io/website/book/codeQuality/nameWell/intermediate/useNameExplain/) try not to use regex1, regex2, etc. Maybe name it todoRegex or just combine lines 27 and 28 into one.

willowisp01 commented on src/main/java/Paimon.java:
```diff
@@ -0,0 +1,180 @@
+        Scanner scanner = new Scanner(System.in);
+        TaskList taskList = FileHandler.loadTaskList();
+        boolean isActive = true;
+        while (isActive) {
```
I think the main loop is pretty long (over 100 lines). Would it be possible to shift the logic elsewhere?

willowisp01 reviewed (commented):
I think the code largely fulfils the naming conventions. Well done!

zhekaiii commented on src/main/java/CommandParser.java:
```diff
@@ -0,0 +1,84 @@
+    public String[] parseInput() throws ChatException {
+        switch (this.type) {
+            case "todo":
+                String regex1 = "^(\\w+)(\\s)(.*)";
+                Pattern pattern1 = Pattern.compile(regex1);
+                Matcher matcher1 = pattern1.matcher(input);
```
Like @willowisp01 mentioned about the naming of the variable, avoid using numbers to distinguish cases.

You can try to combine the lines into one
```suggestion
                Matcher todoMatcher = Pattern.compile("^(\\w+)(\\s)(.*)").matcher(input);
```

zhekaiii commented on src/main/java/FileHandler.java:
```diff
@@ -0,0 +1,79 @@
+package main.java;
+
+import main.java.task.*;
```
Avoid using wildcard imports. You can tweak the settings in IntelliJ IDEA to avoid wildcard imports because they have some default number of before they start converting them to wildcard imports.

You can also try searching for ".*" using your IDE to check for all wildcard imports to make sure they are all removed.

zhekaiii commented on src/main/java/Paimon.java:
```diff
@@ -0,0 +1,180 @@
+
+    }
+
+    private static void greeting() {
```
Method names should be verbs, so an example would be changing this to `printGreeting`. Similar for `goodbye`.

zhekaiii commented on src/main/java/CommandParser.java:
```diff
@@ -0,0 +1,84 @@
+                } else {
+                    throw new ChatException("Input does not match expected format: event <task> /from <time> /to <time>");
+                }
+            case "mark":
+            case "unmark":
```
As stated in the Java coding standard, the explicit `//Fallthrough` comment should be included whenever there is a `case` statement without a break statement.

zhekaiii reviewed (commented):
Overall the Java coding standard has been followed pretty well, although I do notice a lack of Javadocs for classes and methods. Good job!

jovantanyk committed:
Edited gitignore file

jovantanyk committed:
Merged Level-8: Date/Time into master

jovantanyk committed:
Create Command package and modified CommandParser

jovantanyk committed:
Merge A-MoreOOP

jovantanyk committed:
Merge remote-tracking branch 'origin/add-gradle-support' into branch-A-Gradle

jovantanyk committed:
A-Gradle: Edit Gradle Configuration and directories

Shift all files to be under package paimon

jovantanyk committed:
Merge A-Gradle to master branch

jovantanyk committed:
Create CommandParserTest

Create CommandParserTest.java to test CommandParser

jovantanyk committed:
Edit CommandParser to Pass Test Cases

Invalid Regex resulted in false positive test in TodoCommand. Adjust Regex to fix it.

jovantanyk committed:
Add DeadlineCommandTest and Update DateParser

Updated DateParser to properly handle Dates with Time. Added DeadlineCommandTest

jovantanyk committed:
Merge A-JUnit to master branch

jovantanyk committed:
Adjust FileHandler and build.gradle

Build.gradle now creates a paimon.jar, and FileHandler now creates directory if not found

jovantanyk committed:
Merge A-Jar to master branch

jovantanyk committed:
Add FindCommand

Add FindCommand and Tweak UiHandler to accept Strings instead of Tasks

jovantanyk committed:
Add JavaDoc comments

jovantanyk committed:
Merge Level-9: Find to master

jovantanyk committed:
Merge branch 'branch-A-JavaDoc'

jovantanyk committed:
Adjust UiHandler

jovantanyk committed:
style: change methods in UIHandler to be more consistent

Change various response methods to all start with a verb in accordance with java coding standards.

Having a consistent naming style for methods improves code quality and readability

jovantanyk committed:
style: reformat files

Reformat files to be more consistent with indentation and syntax.

Having more consistent indentation in accordance with java conventions allows for better readability

jovantanyk committed:
fix: update DeleteCommand.java

Update DeleteCommand.java to throw an exception instead of using System.out.
UIHandler is now solely handling all output and the program is now consistent

jovantanyk committed:
style: improve imports

Improve imports by reducing the use of wildcard imports. This helps with reducing bloat as only the needed classes are imported

jovantanyk committed:
Merge A-CodingStandard to master branch

Edit and implement various coding standards to adhere to java coding conventions. This includes optimising import statements, indentation and more consistent method names.

jovantanyk committed:
feat: implement checkstyle

Implement checkstyle to ensure consistent syntax and indentation.
Update all files to ensure they follow checkstyle.
Having checkstyle ensures that code is consistent and readable

jovantanyk committed:
Merge A-CheckStyle to master

Implement Checkstyle and ensure that all files follow java conventions. Indentations are now consistent and import statements are ordered accordingly.

jovantanyk committed:
fix: update checkstyle

Fix checkstyle and improper indentation in test files.
Update build.gradle to prepare for JavaFX

jovantanyk committed:
feat: initial javafx commit

Initial JavaFX commit

jovantanyk committed:
feat: convert command line to gui

Convert all CLI based methods to GUI. Create Basic GUI

jovantanyk committed:
feat: add custom images and fix scaling

Adjust Window scaling so that messages and anchor pane can stretch to fill the screen regardless of resizing

jovantanyk committed:
Merge Level-10 to master

Add Basic GUI to app

jovantanyk committed:
feat: add assertions

Add assertions to ensure data integrity

Previously, the code lacked checks for program state assumptions,
potentially leading to subtle bugs.

Introduce assertions to validate program state, such as non-null
inputs and expected conditions, before method executions.

Assertions aid in early detection of logical errors by documenting
and enforcing invariants, improving code robustness and debugging
speed during development and testing phases.

jovantanyk committed:
refactor: update CommandParser.java to adhere to SLAP

Refactor CommandParser.java for SLAP compliance

The CommandParser.java implementation was overly complex, making it difficult to understand, maintain, and extend. The parseInput method especially contained multiple levels of abstraction, mixing command parsing logic with command object creation.

Refactor CommandParser.java to adhere to the Single Level of Abstraction Principle (SLAP). Extract command parsing and object creation into separate, smaller methods within the class. This includes creating individual parse methods for each command type, such as parseTodoCommand, parseDeadlineCommand, etc., and a utility method for regex matching.

This refactoring simplifies the parseInput method, making the CommandParser class easier to read, maintain, and extend. Each command parsing logic is now encapsulated in its own method, improving code modularity and readability. It also makes the class more adaptable to changes in command syntax or the addition of new command types in the future.

jovantanyk committed:
style: removed unused imports

Remove unused imports across the codebase for Checkstyle compliance

The codebase contained numerous unused imports, leading to Checkstyle violations. These unused imports clutter the code, potentially causing confusion and reducing readability.

Remove all unused imports from the codebase to comply with Checkstyle rules. This cleanup involves scrutinizing each file for import statements that are not in use and removing them.

By adhering to Checkstyle guidelines, we enhance code quality and maintainability. Removing unused imports also reduces the risk of naming conflicts and decreases compilation time slightly.

jovantanyk committed:
style: update javadoc comments

Update JavaDoc comments for Git conventions adherence

JavaDoc comments were previously inconsistent with Git conventions, lacking clarity and uniformity across the codebase.

Revise all JavaDoc comments to align with Git conventions. This includes standardizing the format, ensuring concise and informative descriptions, and adding missing tags like @param, @return, and @throws where applicable.

Adopting Git conventions for JavaDoc comments enhances code readability and maintainability, providing clearer documentation for developers and users alike.

jovantanyk committed:
Merge pull request #2 from jovantanyk/branch-A-Assertions

Add assertions to ensure data integrity

Previously, the code lacked checks for program state assumptions,
potentially leading to subtle bugs.

Introduce assertions to validate program state, such as non-null
inputs and expected conditions, before method executions.

Assertions aid in early detection of logical errors by documenting
and enforcing invariants, improving code robustness and debugging
speed during development and testing phases.

jovantanyk committed:
Merge pull request #3 from jovantanyk/branch-A-CodeQuality

Refine code to follow Java conventions and improve JavaDocs

Codebase had mixed adherence to Java conventions, affecting readability
and maintainability. This update standardizes JavaDocs, indentation, and
formatting across all files.

Enhancing consistency with Java coding standards improves readability
and facilitates easier code maintenance. Changes are cosmetic, not
affecting functionality.

This effort aligns the project with best practices, ensuring a cleaner,
more professional codebase appearance.

jovantanyk committed:
feat: add doafter command

Add command for scheduling tasks after specific dates

Currently, users cannot schedule tasks to be automatically started after a specified date, limiting the flexibility and utility of task scheduling within the application.

To enhance user productivity and scheduling capabilities, this commit introduces the doafter command. This command allows users to add tasks that must be initiated after a specific date, expanding the application's functionality to accommodate future-dated tasks.

jovantanyk committed:
feat: add gui resizing and fix testcases

Improve GUI resizing behaviour

Currently, the GUI does not resize smoothly across different screen resolutions and window sizes, leading to a suboptimal user experience with elements often appearing misaligned or improperly scaled.

This limitation hinders the application's usability, especially on devices with varying screen sizes, and detracts from the overall user experience by not accommodating the diverse hardware used by our user base.

Adjust the GUI elements and enhance the resizing logic to ensure a responsive and adaptive user interface. This includes optimizing element proportions and implementing dynamic resizing mechanisms that better handle changes in window size or screen resolution.

jovantanyk committed:
Merged BCD-Extension

Add BCD-Extension and improve testing

jovantanyk committed:
docs:update README.md and Ui.png

jovantanyk committed:
docs:update README.md

jovantanyk committed:
docs: update documentation in README.md

Update Documentation for README.md

Previously, the mark/unmark sections lacked detailed documentation, making it difficult for new developers to understand its purpose and use.

Add comprehensive documentation for feature mark/unmark in the project's documentation section.

The documentation is structured to provide a clear overview of the feature, its usage examples, and configuration options to ensure ease of use and integration into existing projects.

jovantanyk committed:
docs: update documentation for delete task

Update Documentation for Delete Feature

Previously, the Delete Feature sections lacked detailed documentation, making it difficult for new developers to understand its purpose and use.

Add comprehensive documentation for feature Delete Task in the project's documentation section.

The documentation is structured to provide a clear overview of the feature, its usage examples, and configuration options to ensure ease of use and integration into existing projects.

jovantanyk committed:
docs: standardise datetime output

Update documentation to match actual program output

The existing documentation's time format descriptions do not align with the standardized output format used by the program.

Update the documentation to reflect the standardized time format across all examples and explanations.

Standardizing the time format in the documentation ensures consistency with the program's output, reducing confusion and improving user comprehension. This change is made to align documentation closely with the actual behavior of the program, ensuring that users have accurate expectations and understand how to interpret the output correctly.

jovantanyk committed:
fix: update doafter format

Update the doafter command to align with documentation expectations

The doafter command's current implementation deviates from the documented expected output, leading to confusion and mismatches in user expectations.

Modify the doafter command to ensure its output matches the expectations set forth in the documentation.

jovantanyk committed:
fix: remove unnecessary file

Remove unnecessary temporary file paimon.txt from the repo

The file paimon.txt is a temporary storage file that currently exists in the repository.

Remove the unnecessary file paimon.txt to clean up the repository and reduce clutter.

paimon.txt was already included in the .gitignore but for some reason was still being committed
